gaphor/C4Model/toolbox.py

```diff
@@ -0,0 +1,118 @@
+from gaphor.diagram.diagramtools import new_item_factory
+from gaphor.UML import diagramitems as uml_items
+from gaphor.UML.actions.actionstoolbox import actions
+from gaphor.UML.classes.classestoolbox import classes
+from gaphor.UML.interactions.interactionstoolbox import interactions
+from gaphor.UML.states.statestoolbox import states
+
+
+def software_system_config(new_item):
+    default_namespace(new_item)
+    subject = new_item.subject
+    subject.type = gettext("Software System")
+    subject.name = "NewSoftwareSystem"
+
+
+def container_config(new_item):
+    default_namespace(new_item)
+    subject = new_item.subject
+    subject.type = gettext("Container")
+    subject.name = "NewContainer"
+
+
+def component_config(new_item):
+    default_namespace(new_item)
+    subject = new_item.subject
+    subject.type = gettext("Component")
+    subject.name = "NewComponent"
+
+
+c4 = ToolSection(
+    gettext("C4 Model"),
+    (
+        ToolDef(
+            "c4-person",
+            gettext("Person"),
+            "gaphor-c4-person-symbolic",
+            "P",
+            new_item_factory(
+                diagramitems.C4PersonItem,
+                c4model.C4Person,
+                config_func=namespace_config,
+            ),
+        ),
+        ToolDef(
+            "c4-software-system",
+            gettext("Software System"),
+            "gaphor-c4-software-system-symbolic",
+            "s",
+            new_item_factory(
+                diagramitems.C4ContainerItem,
+                c4model.C4Container,
+                config_func=software_system_config,
+            ),
+        ),
+        ToolDef(
+            "c4-container",
+            gettext("Container"),
+            "gaphor-c4-container-symbolic",
+            "o",
+            new_item_factory(
+                diagramitems.C4ContainerItem,
+                c4model.C4Container,
+                config_func=container_config,
+            ),
+        ),
+        ToolDef(
+            "c4-container-database",
+            gettext("Container: Database"),
+            "gaphor-c4-container-database-symbolic",
+            "d",
+            new_item_factory(
+                diagramitems.C4ContainerDatabaseItem,
+                c4model.C4Container,
+                config_func=container_config,
+            ),
+        ),
+        ToolDef(
+            "c4-component",
+            gettext("Component"),
+            "gaphor-c4-component-symbolic",
+            "c",
+            new_item_factory(
+                diagramitems.C4ContainerItem,
+                c4model.C4Container,
+                config_func=component_config,
+            ),
+        ),
+        ToolDef(
+            "c4-dependency",
+            gettext("Dependency"),
+            "gaphor-dependency-symbolic",
+            "<Shift>D",
+            new_item_factory(uml_items.DependencyItem),
+        ),
+    ),
+)
+
+
+c4model_toolbox_actions: ToolboxDefinition = (
+    general_tools,
+    c4,
+    classes,
```

Comment:
Is it common for people to mix C4 with UML? I was expecting that only the general and c4 toolbars would be used, but if this is common then maybe this is a good approach.

Reply:
UML Classes can be used for class modeling. C4 focuses on components and software systems. See https://c4model.com, the Metamodel section.

Reply:
That makes sense, I was just trying to understand the overlap. If it was extremely common to overlap then we should keep them together on the same toolbar, otherwise users could select UML from the dropdown when they want to draw some UML stuff.